Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -542,6 +542,26 @@ variable {R : Type u} [CommRing R] (E : EllipticCurve R)
 def j : R :=
   E.Δ'⁻¹ * E.c₄ ^ 3
 
+theorem j_eq_zero_iff' : E.j = 0 ↔ E.c₄ ^ 3 = 0 := by
+  simp only [EllipticCurve.j, Units.mul_right_eq_zero]
+
+alias ⟨_, j_eq_zero'⟩ := j_eq_zero_iff'
+
+theorem j_eq_zero_iff [IsReduced R] : E.j = 0 ↔ E.c₄ = 0 := by
+  rw [j_eq_zero_iff']
+  exact ⟨fun h ↦ IsNilpotent.eq_zero ⟨_, h⟩, fun h ↦ by simp [h]⟩
+
+theorem j_eq_zero : E.c₄ = 0 → E.j = 0 := fun h ↦ by
```

Comment:
You could use an alias like you did above, but again I don't think we need this, otherwise we'd have lots of directional lemmas for every iff statement.

Reply:
This is not an alias of `j_eq_zero_iff` since it does not need `IsReduced` assumption.

Reply:
> Do you have an example where you need the unidirectional version, and not have a reduced ring?

No, I don't have this example. But I strongly insist to keep this for the sake of completeness.